pkg/validator/expression.go

```diff
@@ -386,3 +386,35 @@ func (h expressionSelectorsMatchesAnything) Validate(rule rulefmt.Rule, promethe
 	}
 	return errs
 }
+
+func newExpressionWithNoMetricName(paramsConfig yaml.Node) (Validator, error) {
+	params := struct{}{}
+	if err := paramsConfig.Decode(&params); err != nil {
+		return nil, err
+	}
+	return &expressionWithNoMetricName{}, nil
+}
+
+type expressionWithNoMetricName struct{}
+
+func (e expressionWithNoMetricName) String() string {
+	return "expression with no metric name"
+}
+
+func (e expressionWithNoMetricName) Validate(rule rulefmt.Rule, prometheusClient *prometheus.Client) []error {
+	if prometheusClient == nil {
+		log.Error("missing the `prometheus` section of configuration for querying prometheus, skipping check that requires it...")
+		return nil
+	}
+	var errs []error
+	names, err := getExpressionMetricsNames(rule.Expr)
+	if err != nil {
+		return []error{err}
+	}
+	for _, s := range names {
+		if s == "" {
+			errs = append(errs, fmt.Errorf("missing metric name for expression `%s`", rule.Expr))
```

Comment:
Maybe instead of appending multiple errors with the same message, we can return just one error with the count of vector selectors missing the metric name?

Or if you would return also the vector selector, you could print out the exact selectors that are missing the metric name :thinking:

Reply:
updated

Reply:
gret! :+1: